src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameRequestStream.cs

```diff
@@ -119,8 +119,19 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             var task = ValidateState(cancellationToken);
             if (task == null)
             {
-                // Needs .AsTask to match Stream's Async method return types
-                return _body.ReadAsync(new ArraySegment<byte>(buffer, offset, count), cancellationToken).AsTask();
+                return _body.ReadAsync(new ArraySegment<byte>(buffer, offset, count), cancellationToken);
+            }
+            return task;
+        }
+
+        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            // REVIEW: Is bufferSize a minimum, maximum or suggestion? Should we validate that it's <= 4096 or >= 4096?
+
```

Comment:
What does `Cap = Math.Max(4096, bufferSize)` do?

I think we should just ignore bufferSize since we will never call WriteAsync with more than 4096 bytes, and I don't feel the need to segment blocks into even smaller chunks unnecessarily.

We can verify it `> 0` 'cause, why not?